From 1e5ddf34fdef7ba1676b2a7e0b5c3335439b04a9 Mon Sep 17 00:00:00 2001
From: Yan Fu <yafu@redhat.com>
Date: Tue, 21 Oct 2025 08:06:17 -0400
Subject: [PATCH 3/7] Fedora: kbs_protocol: default to openssl
Content-type: text/plain

Replace 'rust-crypto' with 'openssl' in default

Use native TLS for openssl

Add "bin" to default to build trustee-attester

Signed-off-by: Uri Lublin <uril@redhat.com>
Signed-off-by: Yan Fu <yafu@redhat.com>
---
 attestation-agent/kbs_protocol/Cargo.toml | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/attestation-agent/kbs_protocol/Cargo.toml b/attestation-agent/kbs_protocol/Cargo.toml
index 0a91ccd..f6b2b51 100644
--- a/attestation-agent/kbs_protocol/Cargo.toml
+++ b/attestation-agent/kbs_protocol/Cargo.toml
@@ -45,7 +45,7 @@ name = "trustee-attester"
 required-features = ["bin"]
 
 [features]
-default = ["background_check", "passport", "rust-crypto", "all-attesters"]
+default = ["background_check", "passport", "openssl", "all-attesters", "bin"]
 
 passport = []
 # Allow to connect Attestation-Agent with TTRPC to get evidence, token, etc.
@@ -69,7 +69,7 @@ se-attester = ["attester/se-attester"]
 tpm-attester = ["attester/tpm-attester"]
 nvidia-attester = ["attester/nvidia-attester"]
 
-rust-crypto = ["reqwest/rustls-tls", "crypto/rust-crypto"]
-openssl = ["reqwest/native-tls-vendored", "crypto/openssl"]
+rust-crypto = []
+openssl = ["reqwest/native-tls", "crypto/openssl"]
 
 bin = ["tokio/rt", "tokio/macros", "clap", "env_logger"]
-- 
2.53.0

